### DISCUSSION TITLE
Co-hosted silo not loading grains

### DISCUSSION THREAD

**Discussion by mariuszklos** [dotnet/orleans#8309] [Q&A]
Hello,

I have aspnet core project with Orleans 7.1. I keep on getting error "ArgumentException: Could not find an implementation for interface"

this is my app init code:
```
var builder = WebApplication.CreateBuilder(args);        
      var host = Host.CreateDefaultBuilder(args)
          .UseOrleans((context, siloBuilder) =>
          {
              siloBuilder.UseLocalhostClustering()
              .AddMemoryGrainStorage("some-name-here")
              .ConfigureLogging(logging =>
              {
                  logging.AddConsole().SetMinimumLevel(LogLevel.Debug);
              }) ;                
          })
          .UseServiceProviderFactory(new AutofacServiceProviderFactory())
          .ConfigureWebHostDefaults(webHostBuilder =>
          {
              webHostBuilder
                  .UseContentRoot(Directory.GetCurrentDirectory())
                  .UseIISIntegration()
                  .UseStartup<Startup>();
          })
          .Build();

      host.Run();
```

and this is how I am trying to get a grain and getting the exception:
`var chatGrain = GrainFactory.GetGrain<IChatGrain>(chatId);`
where
`GrainFactory is IClusterClient`
Grains are in a separate project and added as a reference to the web project.

**Comment by ReubenBond** [comment]
Which packages are you referencing in your application?

* All clients should reference [Microsoft.Orleans.Client](https://nuget.org/packages/Microsoft.Orleans.Client)
* All silos (servers) should reference [Microsoft.Orleans.Server](https://nuget.org/packages/Microsoft.Orleans.Server)
* All other packages should reference [Microsoft.Orleans.Sdk](https://nuget.org/packages/Microsoft.Orleans.Sdk)
